app/uk/gov/hmrc/cgtpropertydisposals/service/BusinessPartnerRecordService.scala

```diff
@@ -65,7 +65,7 @@ class BusinessPartnerRecordServiceImpl @Inject()(connector: BusinessPartnerRecor
     connector.getBusinessPartnerRecord(bprRequest).subflatMap { response =>
       lazy val identifiers =
         List(
-          "id"                -> bprRequest.id.fold(_.value, _.value),
+          "id"                -> bprRequest.foldOnId(_.value, _.value, _.value),
```

Comment:
Since we don't document code, the code itself should be self-documenting. It is difficult to understand what this is achieving here without backtracking and find out the definition of `foldOnId`